cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles.py

```diff
@@ -0,0 +1,91 @@
+import configparser
+import logging
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple
+
+log = logging.getLogger(__name__)
+
+# pylint: disable=too-many-instance-attributes
+@dataclass
+class AzureProfile:
+    name: str
+    subscription_id: str
+    tenant_id: str
+    principal_id: str
+    principal_secret: str
+    location: str
+    resource_group_names: List[str]
+    storage_account_names: List[str]
+
+
+def load_profiles(file_path: str) -> Optional[List[AzureProfile]]:
+    """
+    Return profile list on success
+    Return empty list if file_path doesn't exist or file contains no profiles
+    Return None on file reading error/parsing error
+    """
+
+    config = configparser.ConfigParser()
+    try:
+        # note: config.read silently ignores the fact that file_path doesn't exist, resulting in empty config dict and no error reported
+        config.read(file_path)
+    except configparser.Error:
+        log.exception("Failed to read profiles file '%s'.", file_path)
```

Comment:
I think that logging the same error on multiple layers of function is sub-optimal, because it leads to multiple logs with partial failure information. IMO it is better to re-raise exception with added context information and log it only once at top level (e.g. in load_profiles_or_exit function).

Other issue with that is that this log is not outputted to stderr.
This applies also to log.exception below.

Reply:
ok, will re-raise exception.
callstack is not intended to be printed in console.